src/cow_api.py

```diff
@@ -0,0 +1,189 @@
+    """
+
+    from_: Address
+    sell_token: Address
+    buy_token: Address
+    valid_to: int
+    kind: OrderKind
+
+    sell_amount_before_fee: Optional[int] = None
+    sell_amount_after_fee: Optional[int] = None
+    buy_amount_after_fee: Optional[int] = None
+
+    receiver: Optional[Address] = None
+    app_data: AppData = AppData.zero()
+    partially_fillable: bool = False
+    sell_token_balance: TokenBalance = TokenBalance.ERC20
+    buy_token_balance: TokenBalance = TokenBalance.ERC20
+    price_quality: Optional[PriceQuality] = None
+
+    def to_json(self) -> dict[str, Any]:
+        """
+        Converts a request instance to a dictionary for JSON serialization.
+        """
+
+        # Do some name re-naming. In the future, it may be easier use a library
+        # for serialization like `marshmallow`.
+        value = {
+            "sellToken": self.sell_token.address,
+            "buyToken": self.buy_token.address,
+            "validTo": self.valid_to,
+            "appData": self.app_data.value,
+            "kind": self.kind.value,
+            "partiallyFillable": self.partially_fillable,
+            "sellTokenBalance": self.sell_token_balance.value,
+            "buyTokenBalance": self.buy_token_balance.value,
+            "from": self.from_.address,
+        }
+        if self.receiver is not None:
+            value["receiver"] = self.receiver.address
+        if self.price_quality is not None:
+            value["priceQuality"] = self.price_quality.value
+        if self.kind == OrderKind.SELL and self.sell_amount_before_fee is not None and self.sell_amount_after_fee is None and self.buy_amount_after_fee is None:
+            value["sellAmountBeforeFee"] = str(self.sell_amount_before_fee)
+        elif self.kind == OrderKind.SELL and self.sell_amount_before_fee is None and self.sell_amount_after_fee is not None and self.buy_amount_after_fee is None:
+            value["sellAmountAfterFee"] = str(self.sell_amount_after_fee)
+        elif self.kind == OrderKind.BUY and self.sell_amount_before_fee is None and self.sell_amount_after_fee is None and self.buy_amount_after_fee is not None:
+            value["buyAmountAfterFee"] = str(self.buy_amount_after_fee)
+        else:
+            raise ValueError("must specify exactly one sell/buy amount matching order kind")
+
+        return value
+
+
+@dataclass
+class Quote:
+    """
+    A quoted CowSwap order.
+    """
+    sell_token: Address
+    buy_token: Address
+    receiver: Optional[Address]
+    sell_amount: int
+    buy_amount: int
+    valid_to: int
+    app_data: AppData
+    fee_amount: int
+    kind: OrderKind
+    partially_fillable: bool
+    sell_token_balance: TokenBalance
+    buy_token_balance: TokenBalance
+
+
+@dataclass
+class QuoteResponse:
+    """
+    A CowSwap quote response.
+    """
+    quote: Quote
+    from_: Address
+    expiration: datetime
+
+    @classmethod
+    def from_json(cls, value: dict[str, Any]) -> QuoteResponse:
```

Comment:
uber nit: If you wanted jive with the style of the project, these kinda methods might be better named as to/from_dict.